url30d916877f_shylux_shyview_tgz-pJ8-org_json_TestJ8/src/shyview/TitleInformer.java

```diff
@@ -0,0 +1,82 @@
+package shyview;
+
+import java.awt.Frame;
+import java.util.NoSuchElementException;
+
+public class TitleInformer implements IPicInfo {
+	private static final String NOPICTURES = "No pictures available";
+	
+	private TitleInformer() {};
+	private static TitleInformer _instance;
+	public static TitleInformer getInstance() {
+		if (_instance == null) _instance = new TitleInformer();
+		return _instance;
+	}
+	
+	Frame window;
+	String startTitle;
+	String listName;
+	int listSize;
+	int picPosition;
+	String picName;
+	String process;
+	
+	public void setFrame(Frame fr) {
+		window = fr;
+		startTitle = window.getTitle();
+	}
+
+	private void redraw() {
+		StringBuilder title = new StringBuilder();
+		title.append(startTitle+" - ");
+		if (listName != null && picName != null) {
+			title.append(picPosition+"/"+listSize);
+			title.append(" - ");
+			title.append(listName);
+			title.append(" - ");
+			String filename = picName;
+			if (filename.length() > 20) filename = ".."+filename.substring(filename.length()-20);
+			title.append(filename);
+		} else {
+			title.append(NOPICTURES);
+		}
+
+		if (process != null) {
+			title.append(" - ");
+			title.append(process);
+		}
+		window.setTitle(title.toString());
+	}
+
+	@Override
+	public void update(IPicList list) {
+		listName = list.getName();
+		listSize = list.size();
+		try {
+			picName = list.current().getName();
+			picPosition = list.getIndex()+1;
+		} catch (NoSuchElementException e) {}
+		redraw();
+	}
+
+	@Override
+	public void clear() {
+		listName = null;
+		picName = null;
+		redraw();
+	}
+
+	@Override
+	public void pushProcess(String info) {
+		if (process == info) return;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.